picard/log.py

```diff
@@ -56,67 +54,8 @@ def debug_mode(enabled):
     'TailLogTuple', ['pos', 'message', 'level'])
 
 
-def _DummyFn(*args, **kwargs):
-    """Placeholder function.
-
-    Raises:
-        NotImplementedError
-    """
-    _, _ = args, kwargs
-    raise NotImplementedError()
-
-
-# _srcfile is used when walking the stack to check when we've got the first
-# caller stack frame, by skipping frames whose filename is that of this
-# module's source. It therefore should contain the filename of this module's
-# source file.
-_srcfile = os.path.normcase(_DummyFn.__code__.co_filename)
-if hasattr(sys, '_getframe'):
-    def currentframe():
-        return sys._getframe(3)
-else:  # pragma: no cover
-    def currentframe():
-        """Return the frame object for the caller's stack frame."""
-        try:
-            raise Exception
-        except Exception:
-            return sys.exc_info()[2].tb_frame.f_back
-
-
 class OurLogger(logging.getLoggerClass()):
```

Comment:
I think you can get rid of this class completely - at least doing that didn't change any of the messages during picard startup with debug mode enabled.